fineract-core/src/main/java/org/apache/fineract/infrastructure/event/external/jobs/SendAsynchronousEventsTasklet.java

```diff
@@ -104,14 +111,40 @@ private void markEventsAsSent(List<Long> eventIds) {
         // Partitioning dataset to avoid exception: PreparedStatement can have at most 65,535 parameters
         final int partitionSize = fineractProperties.getEvents().getExternal().getPartitionSize();
         List<List<Long>> partitions = Lists.partition(eventIds, partitionSize);
+        final ThreadPoolTaskExecutor executor = getThreadPoolTaskExecutor();
+        final FineractContext context = ThreadLocalContextUtil.getContext();
+        List<Future<?>> tasks = new ArrayList<>();
         partitions //
                 .forEach(partitionedEventIds -> {
-                    measure(() -> {
-                        repository.markEventsSent(partitionedEventIds, sentAt);
-                    }, timeTaken -> {
-                        log.debug("Took {}ms to update {} events", timeTaken.toMillis(), partitionedEventIds.size());
-                    });
+                    tasks.add(executor.submit(() -> {
+                        measure(() -> {
+                            ThreadLocalContextUtil.init(context);
+                            repository.markEventsSent(partitionedEventIds, sentAt);
+                        }, timeTaken -> {
+                            log.debug("Took {}ms to update {} events", timeTaken.toMillis(), partitionedEventIds.size());
+                        });
+                    }));
                 });
+        tasks.forEach(task -> {
```

Comment:
The problem is the `task.get` is blocking call and we dont really need the result...